src/cargo/ops/registry.rs

```diff
@@ -390,6 +394,22 @@ pub fn configure_http_handle(config: &Config, handle: &mut Easy) -> CargoResult<
     Ok(())
 }
 
+/// Find an override from config for curl low-speed-limit option, otherwise use default value
+fn low_speed_limit(config: &Config) -> CargoResult<Option<u32>> {
+    if let Some(s) = config.get::<Option<u32>>("http.low-speed-limit")? {
+        return Ok(Some(s));
+    }
+    Ok(Some(10))
```

Comment:
Since this always returns `Some`, perhaps this could return `CargoResult<u32>` and use `.unwrap_or(10)` internally?

Reply:
Removed the Option, but instead of unwrap_or I've set the default in the function. This way, everything is in one place.